Tensile/Source/lib/include/Tensile/Utils.hpp

```diff
@@ -37,6 +37,59 @@
 namespace Tensile
 {
 
+    // Type wrapper that can be copied or assigned to in a threadsafe manner. Value cannot be modified
+    template<typename T>
+    struct ThreadSafeValue
+    {
+    private:
+        mutable std::mutex access;
+        T                  value;
+    public:
+        ThreadSafeValue()
+        {
+
+        }
+
+        ThreadSafeValue(const ThreadSafeValue<T>& other)
```

Comment:
You've redefined the copy constructor and the assignment operator, but have not defined a destructor. This breaks the [rule of three](https://en.cppreference.com/w/cpp/language/rule_of_three).